chips/lowrisc/src/i2c.rs

```diff
@@ -0,0 +1,275 @@
+        SCL_INTERFERENCE OFFSET(5) NUMBITS(1) [],
+        SDA_INTERFERENCE OFFSET(6) NUMBITS(1) [],
+        STRETCH_TIMEOUT OFFSET(7) NUMBITS(1) [],
+        SDA_UNSTABLE OFFSET(8) NUMBITS(1) []
+    ],
+    CTRL [
+        ENABLEHOST OFFSET(0) NUMBITS(1) []
+    ],
+    STATUS [
+        FMTFULL OFFSET(0) NUMBITS(1) [],
+        RXFULL OFFSET(1) NUMBITS(1) [],
+        FMTEMPTY OFFSET(2) NUMBITS(1) [],
+        HOSTIDLE OFFSET(3) NUMBITS(1) [],
+        TARGETIDLE OFFSET(4) NUMBITS(1) [],
+        RXEMPTY OFFSET(5) NUMBITS(1) []
+    ],
+    RDATA [
+        RDATA OFFSET(0) NUMBITS(8) []
+    ],
+    FDATA [
+        FBYTE OFFSET(0) NUMBITS(8) [],
+        START OFFSET(8) NUMBITS(1) [],
+        STOP OFFSET(9) NUMBITS(1) [],
+        READ OFFSET(10) NUMBITS(1) [],
+        RCONT OFFSET(11) NUMBITS(1) [],
+        NAKOK OFFSET(12) NUMBITS(1) []
+    ],
+    FIFO_CTRL [
+        RXRST OFFSET(0) NUMBITS(1) [],
+        FMTRST OFFSET(1) NUMBITS(1) [],
+        RXILVL OFFSET(2) NUMBITS(3) [
+            RXLVL1 = 0,
+            RXLVL4 = 1,
+            RXLVL8 = 2,
+            RXLVL16 = 3,
+            RXLVL30 = 4
+        ],
+        FMTILVL OFFSET(5) NUMBITS(3) [
+            FMTLVL1 = 0,
+            FMTLVL4 = 1,
+            FMTLVL8 = 2,
+            FMTLVL16 = 3,
+            FMTLVL30 = 4
+        ]
+    ],
+    FIFO_STATUS [
+        FMTLVL OFFSET(0) NUMBITS(6) [],
+        RXLVL OFFSET(16) NUMBITS(6) []
+    ],
+    OVRD [
+        TXOVRDEN OFFSET(0) NUMBITS(1) [],
+        SCLVAL OFFSET(1) NUMBITS(1) [],
+        SDAVAL OFFSET(2) NUMBITS(1) []
+    ],
+    VAL [
+        SCL_RX OFFSET(0) NUMBITS(1) [],
+        SDA_RX OFFSET(1) NUMBITS(1) []
+    ],
+    TIMING0 [
+        THIGH OFFSET(0) NUMBITS(16) [],
+        TLOW OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING1 [
+        T_R OFFSET(0) NUMBITS(16) [],
+        T_F OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING2 [
+        TSU_STA OFFSET(0) NUMBITS(16) [],
+        THD_STA OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING3 [
+        TSU_DAT OFFSET(0) NUMBITS(16) [],
+        THD_DAT OFFSET(16) NUMBITS(16) []
+    ],
+    TIMING4 [
+        TSU_STO OFFSET(0) NUMBITS(16) [],
+        T_BUF OFFSET(16) NUMBITS(16) []
+    ],
+    TIMEOUT_CTRL [
+        VAL OFFSET(0) NUMBITS(31) [],
+        EN OFFSET(31) NUMBITS(1) []
+    ]
+];
+
+pub struct I2c<'a> {
+    registers: StaticRef<I2cRegisters>,
+
+    master_client: OptionalCell<&'a dyn hil::i2c::I2CHwMasterClient>,
+}
+
+impl I2c<'a> {
+    pub const fn new(base: StaticRef<I2cRegisters>) -> I2c<'a> {
+        I2c {
+            registers: base,
+            master_client: OptionalCell::empty(),
+        }
+    }
+
+    pub fn handle_interrupt(&self) {
+        unimplemented!();
+    }
+
+    fn timing_parameter_init(&self) {
+        let regs = self.registers;
+
+        // Setup the timing variables
+        // This is taken from the OT documentation example
+        // Assume a desired datarate of 1 Mbaud (the bus maximum) for an SCL
```

Comment:
Hmm.. 400 KHz is standard for I2C (well, strictly speaking 100 KHz is standard, 400 KHz is "fast mode", 1 MHz is "fast mode plus", and 3.2 MHz is "ultra fast"), but in practice unlabeled I2C is usually "fast mode".

Does the I2C HIL not expose bus speed control? (It probably should..). But I think 400 KHz would be a better default for common-case compatibility.